### ISSUE TITLE
Video Captions: Styling in editor

### ISSUE THREAD

**Issue by bmattb** [GoogleForCreators/web-stories-wp#4899]
## Feature Description

EDIT: The description below is out-dated due to reducing of the scope of the original issue.
What remains to be done here is the following:
- Ensure that background and regular video design panels display the same way
- Ensure that captions remain when a video is set to background
See Pascal's [comment below](https://github.com/google/web-stories-wp/issues/4899#issuecomment-721073194).

OUT-DATED:
#4189 (PR: #4615) added the foundation for the new Subtitles/Captions feature to video elements. However, the [current version](https://www.youtube.com/watch?v=OQFlz0izD_E&feature=youtu.be) does not look like [the desired design](https://github.com/google/web-stories-wp/issues/4189#issuecomment-705524670) yet ([view in Figma](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories---Post-Stable?node-id=1069%3A230257)).

This ticket is about updating the styling of the captions feature to match expected design.

**Note:** as discussed in #4189, we'll use the WordPress media modal for now for uploading the files. That means no inline progress bars or error messages.

Thus, I expect the effort required for this styling to be rather small.

_However_, depending on the progress on #4749, this could also mean updating the dropdown component to use the generalized version with autocomplete etc.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by miina** [comment]
@o-fernandez  Would search be useful here?

It looks like it might make sense to just use the generalized component from #4749 here (in review currently), especially if we can enable search, too.

**Comment by miina** [comment]
@swissspidy I've read through the discussion in https://github.com/google/web-stories-wp/issues/4189 and if I understand correctly then the description of this issue is out of date:
- Since we only support one caption now, being blocked by #4749 and using a dropdown is not relevant anymore
- The only thing needed in this issue is verifying the style matches Figma, however, since we're only allowing one file and are using the WP media modal for uploading the files then it's basically just verifying the one button for uploading and input for removing the selected captions matches the style?

However, for the style to be consistent with the rest of the design panel, we are still using the general input components everywhere until restyling all the components together for 1.3 (#4830 - [see line 21](https://docs.google.com/spreadsheets/d/10DZZrWJdkjcJXD1vL0VG0PqRhsIqaHkacF_ujOdEwJc/edit#gid=0)), we wouldn't update only one input/button to match the new style and leave the rest as they were.

This means that I'm actually not really sure what there is to do currently within the scope of this ticket at all. The input and the button already seem to match the style of the rest of the design panel. Thoughts?

**Comment by swissspidy** [comment]
@miina Yeah you're right; we went back and forth on #4189 and ended up with a much smaller scope. This indeed led to a more consistent styling.

So yeah, not much to do here anymore!

The things that should/could be improved though IMO, possibly in this ticket:

A) Order of panels is inconsistent between background videos and regular videos:

<img width="300" alt="Screenshot 2020-11-03 at 12 53 34" src="https://user-images.githubusercontent.com/841956/97982155-a34d7700-1dd3-11eb-93f5-085ec8836516.png">
<img width="391" alt="Screenshot 2020-11-03 at 12 50 57" src="https://user-images.githubusercontent.com/841956/97982164-a6486780-1dd3-11eb-8a9b-8ab2cdcd7418.png">

B) Setting a video as background video removes the caption

**Comment by csossi** [comment]
Verified in QA

**Comment by samitron7** [comment]
Failed in UAT. Set a video with caption to the bg and note that the caption is clipped on the left and right side. Caption should be fixed to the width of the page and not video. 
<img width="495" alt="Screen Shot 2020-11-30 at 11 32 11 PM" src="https://user-images.githubusercontent.com/10889431/100710653-dc422280-3364-11eb-9142-5f912a2f79cb.png">

Also, can we easily change the styling of the caption?

**Comment by miina** [comment]
@samitron7 
This issue was actually addressing two different things (the name of the issue is a bit deceiving):
- Ensure that background and regular video design panels display in the same order (previously the video caption panel was in a different place for background and regular video)
- Ensure that captions remain when a video is set to background (previously it disappeared after setting a video as background)

We should probably create a new issue to investigate the bug you found.

**Comment by swissspidy** [comment]
> Failed in UAT. Set a video with caption to the bg and note that the caption is clipped on the left and right side. Caption should be fixed to the width of the page and not video.

As @miina said, this ticket is about updates in the editor, nothing frontend-related.

We have #5315 for that specific issue you mention here. It's actually partially a format limitation as well, see https://github.com/ampproject/amphtml/issues/15091#issuecomment-491976689

> Also, can we easily change the styling of the caption?

The styling options for captions are [very limited](https://developer.mozilla.org/en-US/docs/Web/Guide/Audio_and_video_delivery/Adding_captions_and_subtitles_to_HTML5_video#Styling_the_displayed_subtitles), and basically only allows changing font & color, nothing else.

**Comment by swissspidy** [comment]
Closing since #5315 exists for the (unrelated) issue raised here.
